module/zfs/zvol.c

```diff
@@ -1358,6 +1358,8 @@ zvol_open(struct block_device *bdev, fmode_t flag)
 	mutex_exit(&zv->zv_state_lock);
 	if (drop_suspend)
 		rw_exit(&zv->zv_suspend_lock);
+	if (error)
+		schedule();
```

Comment:
Let's restrict this to the `-ERESTARTSYS` case.  This makes it clear exactly why this additional schedule call is here, all other error values should be returned to `sys_open()`.

```c
if (error == -ERESTARTSYS)
       schedule();
```